graphics/blit/blit-neon.cpp

```diff
@@ -324,6 +324,71 @@ void BlendBlit::blitNEON(Args &args, const TSpriteBlendMode &blendMode, const Al
 	blitT<BlendBlitImpl_NEON>(args, blendMode, alphaType);
 }
 
+void fastBlitNEON_XRGB1555_RGB565(byte *dst, const byte *src,
+                  const uint dstPitch, const uint srcPitch,
+                  const uint w, const uint h) {
+	const uint srcDelta = (srcPitch - w * 2);
+	const uint dstDelta = (dstPitch - w * 2);
+
+	const uint16 *src_ptr = (const uint16 *)src;
+	uint16 *dst_ptr = (uint16 *)dst;
+	uint16x4_t pixels;
+	uint16 tempPixels[4];
+
+	for (uint y = h; y > 0; --y) {
+		for (int x = w; x > 0; x -= 4) {
+			src_ptr = (const uint16 *)src;
+			dst_ptr = (uint16 *)dst;
+
+			uint nb_pixels = MIN(4, x);
+
+			// Load pixels to NEON
+			if (nb_pixels == 4) {
+				pixels = vld1_u16(src_ptr);
+			} else {
+				tempPixels[0] = *src_ptr;
+				if (nb_pixels == 3) {
+					tempPixels[1] = *(src_ptr + 1);
+					tempPixels[2] = *(src_ptr + 2);
+					tempPixels[3] = 0;
+				} else if (nb_pixels == 2) {
+					tempPixels[1] = *(src_ptr + 1);
+					tempPixels[2] = 0;
+					tempPixels[3] = 0;
+				} else {
+					tempPixels[1] = 0;
+					tempPixels[2] = 0;
+					tempPixels[3] = 0;
+				}
+				pixels = vld1_u16(tempPixels);
+			}
+
+			// Convert from XRGB1555 to RGB565
+			// Here we do : ((pixels & 0x7FC0) << 1) | (pixels & 0x003F)
```

Comment:
This should be factored in as well to expand the green component:
```suggestion
			// Here we do : ((pixels & 0x7FC0) << 1) | (pixels & 0x003F) | ((pixel 0x0200) >> 4)
```

Reply:
I made a mistake, I wanted to use this : `((pixels & 0x7FE0) << 1) | (pixels & 0x003F)`
As we use this to have a full white 0xFFFF, this should expand the same way minus 3 operations.

Reply:
Duplicating the top bit would be closer to the default implementation (see https://github.com/scummvm/scummvm/blob/master/graphics/pixelformat.h#L91), and should provide a bit more even rounding.

Reply:
Ok, I changed it